Move test db seeding from before:suite to before:each in spec_helper

Steveo1485 opened the pull request.

Steveo1485 committed:
Move test db seeding from before:suite to before:each in spec_helper

smart-alek commented:
Wouldn''t it be a better idea to keep the seeding before the suite and change testing so that the big clean happens after the whole suite is finished? That's a LOT of Pokemon to reload for each test. How long does it take to run those tests right now?

Steveo1485 committed:
Moved seeding back to before suite, changed database to clean after suite, updated trainer_spec to test trainer creation against last Trainer id and name

Steveo1485 committed:
Update trainer name in journey start test to confirm passing test

Steveo1485 commented:
It didn't take long, but could run into problems later on. So I've moved seeding back to before the test suite, but changed the database cleaning to not happen until after the entire suite, instead of after each test. I had to update the "should allow user to start journey" test in the trainer_spec to test that the path equals the last Trainer ID (at the point of trainer creation) and also added line to test that last Trainer name equals the correct name. Changed the name for the "should allow user to start journey" test to Lt. Surge to confirm test was comparing correct trainer.
